### PR TITLE
Add owner as parameter

### PR DESCRIPTION
**Pull request by erwan-joly**
windows_10_x64

- [x] Can't check both Only and Exclude
- [x] Parse authors

### CODE REVIEW

File: src/option/option-window.tsx [hunk 1]
```diff
@@ -100,18 +119,25 @@ export default class OptionWindow extends React.Component {
 	handleSubmit = (event) => {
 		event.preventDefault();
 		let flags: string[];
+		let authors: string[];
 		let frequency: number;
 		let repoUrl: string;
+		let ExcludeOrOnly: boolean;
```
Comment (chris-findlay) [inline src/option/option-window.tsx]: Why is this capitalised?

File: src/option/option-window.tsx [hunk 2]
```diff
@@ -100,18 +119,25 @@ export default class OptionWindow extends React.Component {
...
+
 		frequency = event.target.elements.frequencyInput.value;
 		flags = event.target.elements.flagInput.value.split(';').map(item => item.trim()).filter((e) => e);
+		authors = event.target.elements.authorInput.value.split(';').map(item => item.trim()).filter((e) => e);
 		repoUrl = event.target.elements.urlInput.value;
+		ExcludeOrOnly = event.target.elements.ExcludeOrOnly.value === 'Exclude' ? false : true;
```
Comment (chris-findlay) [inline src/option/option-window.tsx]: Why not `event.target.elements.ExcludeOrOnly.value !== 'Exclude'`?
Also if this is a boolean, it'd be better named for what it is when true, as it's not clear which is which at the moment.

File: src/pull-request-poller/pull-request-poller.tsx [hunk 3]
```diff
@@ -30,6 +31,17 @@ export default class PullRequestPoller extends React.Component<FlagToListenTo, {
...
+								let author = pullrequests[i].parentElement.querySelector('a.muted-link').innerHTML;
+								if (this.props.authors.length > 0) {
+									if (this.props.ExcludeOrOnly && !this.props.authors.includes(author)) {
+										//doesn't show the notification if it's not matching only rules
+										continue;
+									}
+									if (!this.props.ExcludeOrOnly && this.props.authors.includes(author)) {
+										//doesn't show the notification if author is excluded
+										continue;
+									}
```
Comment (chris-findlay) [inline src/pull-request-poller/pull-request-poller.tsx]: Consider XOR: `if (this.props.ExcludeOrOnly ^ this.props.authors.includes(author)) {` to consolidate both `if`s, but keep both comments.
Also, I tend to write comments as proper sentences with capitals and everything, including a space after the `//`, and reserve writing `//` not followed by a space for commented out code, so I can read 1 char and tell which is which.  Just a suggestion.

Comment (erwan-joly) [reply src/pull-request-poller/pull-request-poller.tsx]: XOR maybe a bit more difficult to read unfortunately :/

Comment (chris-findlay) [reply src/pull-request-poller/pull-request-poller.tsx]: If everyone says that then no one learns xor :-p

File: src/option/option-window.tsx [hunk 4]
```diff
@@ -60,6 +63,22 @@ export default class OptionWindow extends React.Component {
...
 	}
 
+	UnRequireInput = () => {
+		const author = document.getElementById('authorInput') as HTMLInputElement;
+		const flag = document.getElementById('flagInput') as HTMLInputElement;
+		if (!author.value) {
+			flag.required = true;
+		} else {
+			flag.required = false;
+		}
```
Comment (chris-findlay) [inline src/option/option-window.tsx]: `flag.required = !author.value`; similarly below.

Comment (erwan-joly) [reply src/option/option-window.tsx]: will apply in the next PR

**Pull request merged**